### PR TITLE
Added Consul module

### PR DESCRIPTION
**Pull request by Chris-AS1**
## What does this PR do?
This PR adds the Consul module. `modulegen` was used as starting point, on top of which a couple of methods were added.

## Why is it important?
This module helps integrate testing with Consul which can be useful to verify the correct behavior under different scenarios.

## How to test this PR
Tests are included within the module itself.

### CODE REVIEW

File: modules/consul/examples_test.go
Comment (mdelapenya) [inline modules/consul/examples_test.go:12]: Let's match the placeholder used in the docs (https://github.com/testcontainers/testcontainers-go/pull/2194/files#diff-722ef8837b8f820bf7c2f317340222a02ce1665a06b5f551bb065e6b5c5b348cR20)
```suggestion
	// runConsulContainer {
```

File: modules/consul/examples_test.go
Comment (mdelapenya) [inline modules/consul/examples_test.go:19]: In https://github.com/testcontainers/testcontainers-go/pull/2193 we replaced panics with log.Fatalf.

Could you update all of them? 🙏  Please check out the golint exclusions for after-defer exits in that PR

**Review by mdelapenya — COMMENTED** [review]
This PR looks good to me, thanks for your time contributing the module 👏 

Could please remove the `examples/consult` directory and all the references to it?:
- dependabot
- ci.yml
- vscode workspace
- mkdocs
- sonar.properties

Once there, I think we can merge it. Thanks!

File: modules/consul/consul.go
Comment (mdelapenya) [inline modules/consul/consul.go:44]: Not a blocker, just a question: do we want to accept a simple string or an io.Reader to allow reading configs from file?

Comment (Chris-AS1) [reply modules/consul/consul.go:44]: I like the idea of having an io.Reader here. Perhaps I should add another Option that implements it? I wouldn't be too sure about replacing the string parameter altogether as it might be useful to test a certain parameters more quickly. What would you say?

Comment (mdelapenya) [reply modules/consul/consul.go:44]: We could have a `WithLocalConfigReader(r io.Reader)` as a generalisation of the read process, keeping the string specialisation.

Comment (Chris-AS1) [reply modules/consul/consul.go:44]: Another option might be something like `WithConfigFile(path string)`, and then map the file to the container. Would you say that's preferred?

Comment (mdelapenya) [reply modules/consul/consul.go:44]: We usually have that in database modules for passing config files, so I think it'd make sense to add it

**Comment by Chris-AS1** [comment]
> This PR looks good to me, thanks for your time contributing the module 👏
> 
> Could please remove the `examples/consult` directory and all the references to it?:
> 
> * dependabot
> * ci.yml
> * vscode workspace
> * mkdocs
> * sonar.properties
> 
> Once there, I think we can merge it. Thanks!

I noticed there are references to `examples/consul` inside `RELEASING.md` too.  Want me to remove those?

**Comment by mdelapenya** [comment]
> I noticed there are references to `examples/consul` inside `RELEASING.md` too. Want me to remove those?

Yes please, I can do the toil for you given the case, so feel free to ping me to perform them if needed

File: RELEASING.md
Comment (Chris-AS1) [inline RELEASING.md:1]: @mdelapenya I'm unsure about what to do on line 39-40. Should those be removed too?

Comment (mdelapenya) [reply RELEASING.md:1]: Do not worry about that: it's an output of the release script, more for maintainers. I'd say it's fine with removing or keeping the entries, as at some point, the log will be outdated, so I should update it.

**Review by mdelapenya — APPROVED** [review]
LGTM! Thanks for this new module 🚀

File: modules/consul/consul.go
Comment (codefromthecrypt) [inline modules/consul/consul.go:72]: note this introduced a flake. The former example was waiting for the listen port (as are most ad-hoc usages on github). This passes before the api is ready failing tests. I think it should be reverted to what the example used:


```
		testcontainers.WithWaitStrategy(wait.ForListeningPort("8500/tcp")),
```

Comment (mdelapenya) [reply modules/consul/consul.go:72]: @codefromthecrypt I went ahead and created an issue for it. Would you like to work on the fix?

Comment (codefromthecrypt) [reply modules/consul/consul.go:72]: actually yes thanks!

Comment (Chris-AS1) [reply modules/consul/consul.go:72]: Completely an oversight on my part, thank you for noticing!

Comment (mdelapenya) [reply modules/consul/consul.go:72]: No worries @Chris-AS1 shit happens 😄 Thankfully this community has eyes on every place 💪

**Pull request merged**
